ntbs-service/Services/LegacySearchService.cs

```diff
@@ -89,6 +92,26 @@ public async Task<(IEnumerable<NotificationBannerModel> notifications, int count
             return (notificationBannerModels, count);
         }
 
+        public async Task<NotificationBannerModel> SearchByIdAsync(string notificationId)
+        {
+            string fullQuery = SelectQueryStart + SelectByIdCondition;
+            dynamic result;
+            
+            using (var connection = new SqlConnection(connectionString))
+            {
+                connection.Open();
+
+                result = (await connection.QueryAsync(fullQuery, new { id = notificationId})).FirstOrDefault();
+            }
+
+            if (result == null)
+            {
+                return null;
+            }
+
+            return ((NotificationBannerModel) AsNotificationBannerAsync(result).Result);
```

Comment:
I don't think `Result` here is necessary? We should be able to use `await` instead?